### PR TITLE
Change how the default class translation files path is constructed

### PR DESCRIPTION
**Pull request by marko-bekhta**
Hey @jamezp 

with the most recent release of `[plexus-compiler](https://github.com/codehaus-plexus/plexus-compiler)` and in particular after this fix: https://github.com/codehaus-plexus/plexus-compiler/pull/350  we've started to see build failures with:

```java
[ERROR] Unknown source: Cannot read org.hibernate.search.util.impl.test.logging package files, cause : java.io.FileNotFoundException: Resource does not exist : CLASS_OUTPUT/org.hibernate.search.util.impl.test.logging/Log
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.BatchFilerImpl.getResource(BatchFilerImpl.java:173)
[ERROR] 	at org.jboss.logging.processor.apt.TranslationClassGenerator.findTranslationFiles(TranslationClassGenerator.java:152)
[ERROR] 	at org.jboss.logging.processor.apt.TranslationClassGenerator.processTypeElement(TranslationClassGenerator.java:115)
[ERROR] 	at org.jboss.logging.processor.apt.LoggingToolsProcessor.doProcess(LoggingToolsProcessor.java:213)
[ERROR] 	at org.jboss.logging.processor.apt.LoggingToolsProcessor.process(LoggingToolsProcessor.java:167)
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:141)
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:123)
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:172)
[ERROR] 	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:952)
[ERROR] 	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:449)
[ERROR] 	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:425)
[ERROR] 	at org.eclipse.jdt.internal.compiler.batch.Main.performCompilation(Main.java:4753)
[ERROR] 	at org.eclipse.jdt.internal.compiler.tool.EclipseCompilerImpl.call(EclipseCompilerImpl.java:101)
[ERROR] 	at org.eclipse.jdt.internal.compiler.tool.EclipseCompiler$1.call(EclipseCompiler.java:196)
[ERROR] 	at org.codehaus.plexus.compiler.eclipse.EclipseJavaCompiler.performCompile(EclipseJavaCompiler.java:309)
[ERROR] 	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1188)
[ERROR] 	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:212)
[ERROR] 	at com.gradle.maven.cache.extension.d.c.a(SourceFile:27)
[ERROR] 	at com.gradle.maven.cache.extension.d.n.a(SourceFile:23)
[ERROR] 	at com.gradle.maven.cache.extension.d.h.a(SourceFile:28)
[ERROR] 	at com.gradle.maven.cache.extension.d.m.a(SourceFile:27)
[ERROR] 	at com.gradle.maven.cache.extension.d.a.c(SourceFile:115)
[ERROR] 	at com.gradle.maven.cache.extension.d.a.a(SourceFile:61)
[ERROR] 	at com.gradle.maven.cache.extension.d.e.a(SourceFile:27)
[ERROR] 	at com.gradle.maven.cache.extension.d.l.a(SourceFile:17)
[ERROR] 	at com.gradle.maven.cache.extension.d.d.a(SourceFile:42)
[ERROR] 	at com.gradle.maven.cache.extension.d.b.a(SourceFile:26)
[ERROR] 	at com.gradle.maven.cache.extension.d.f$1.run(SourceFile:35)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(SourceFile:29)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(SourceFile:26)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(SourceFile:66)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(SourceFile:59)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(SourceFile:157)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(SourceFile:59)
[ERROR] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(SourceFile:47)
[ERROR] 	at com.gradle.maven.cache.extension.d.f.a(SourceFile:31)
[ERROR] 	at com.gradle.maven.cache.extension.d.k.a(SourceFile:76)
[ERROR] 	at com.gradle.maven.cache.extension.h.b.lambda$createProxy$0(SourceFile:79)
[ERROR] 	at jdk.proxy10/jdk.proxy10.$Proxy78.execute(Unknown Source)
[ERROR] 	at com.gradle.maven.scan.extension.internal.d.a.executeMojo(SourceFile:114)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
[ERROR] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:52)
[ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:161)
[ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:73)
[ERROR] 
```

e.g. https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/main/1041/pipeline/121 
Let me know if the change makes sense to you. Thanks!

### CODE REVIEW

**Comment by marko-bekhta** [comment]
Hi James, just a reminder about this one in case it got lost  🙈

**Comment by jamezp** [comment]
My apologies @marko-bekhta. I'll try to look at it this week :)

**Comment by jamezp** [comment]
I've filed https://issues.redhat.com/browse/LOGTOOL-158 for this. I will have a look at it tomorrow. I want to think about this. I need to refresh my memory a bit too :)

File: processor/src/main/java/org/jboss/logging/processor/apt/TranslationClassGenerator.java [hunk 1]
```diff
@@ -146,10 +147,18 @@ private List<File> findTranslationFiles(final MessageInterface messageInterface)
         if (translationFilesPath != null) {
             classTranslationFilesPath = translationFilesPath + packageName.replace('.', File.separatorChar);
 
-            //By default use the class output folder
+            //By default, use the class output folder
         } else {
-            FileObject fObj = processingEnv.getFiler().getResource(StandardLocation.CLASS_OUTPUT, packageName, interfaceName);
-            classTranslationFilesPath = fObj.toUri().getPath().replace(interfaceName, "");
+            // Create some random name:
+            String relativeName = interfaceName + UUID.randomUUID();
```
Comment (jamezp) [inline processor/src/main/java/org/jboss/logging/processor/apt/TranslationClassGenerator.java:153]: What's weird to me is the exception is `java.io.FileNotFoundException`, yet using a random file name makes it work.

**Comment by marko-bekhta** [comment]
Hey James 😃 
I've pushed one more commit that adds an eclipse compiler profile so it is easier to debug the problem. I've been using:
```bash
mvnDebug clean install -Pcompiler-eclipse
```
and attaching a remote debugger in an IDE.
I had to move out the exception classes out of the interfaces as otherwise using an eclipse compiler leads to:
```
ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:testCompile (default-testCompile) on project jboss-logging-processor: Compilation failure
[ERROR] Unknown source: java.lang.UnsupportedOperationException: Instantiating a Nested type CustomException of Reference of type ValidMessages as a class
[ERROR] 	at org.jboss.jdeparser.AbstractJType._new(AbstractJType.java:79)
[ERROR] 	at org.jboss.logging.processor.generator.model.ImplementationClassModel.constructReturnType(ImplementationClassModel.java:516)
[ERROR] 	at org.jboss.logging.processor.generator.model.ImplementationClassModel.createReturnType(ImplementationClassModel.java:409)
[ERROR] 	at org.jboss.logging.processor.generator.model.ImplementationClassModel.createBundleMethod(ImplementationClassModel.java:264)
[ERROR] 	at org.jboss.logging.processor.generator.model.MessageBundleImplementor.generateModel(MessageBundleImplementor.java:74)
[ERROR] 	at org.jboss.logging.processor.generator.model.ClassModel.generateAndWrite(ClassModel.java:126)
[ERROR] 	at org.jboss.logging.processor.apt.ImplementationClassGenerator.processTypeElement(ImplementationClassGenerator.java:58)
[ERROR] 	at org.jboss.logging.processor.apt.LoggingToolsProcessor.doProcess(LoggingToolsProcessor.java:213)
[ERROR] 	at org.jboss.logging.processor.apt.LoggingToolsProcessor.process(LoggingToolsProcessor.java:165)
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:141)
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:123)
[ERROR] 	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:172)
[ERROR] 	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:952)
[ERROR] 	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:449)
[ERROR] 	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:425)
[ERROR] 	at org.eclipse.jdt.internal.compiler.batch.Main.performCompilation(Main.java:4753)
[ERROR] 	at org.eclipse.jdt.internal.compiler.tool.EclipseCompilerImpl.call(EclipseCompilerImpl.java:101)
[ERROR] 	at org.eclipse.jdt.internal.compiler.tool.EclipseCompiler$1.call(EclipseCompiler.java:196)
[ERROR] 	at org.codehaus.plexus.compiler.eclipse.EclipseJavaCompiler.performCompile(EclipseJavaCompiler.java:309)
[ERROR] 	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1140)
[ERROR] 	at org.apache.maven.plugin.compiler.TestCompilerMojo.execute(TestCompilerMojo.java:199)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
[ERROR] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)

```
And I didn't want to dig into `org.jboss.jdeparser` and make it like eclipse compiler as well 😃.

We can drop this change once we are happy with the fix or however you'd like to proceed.

File: processor/src/main/java/org/jboss/logging/processor/apt/TranslationClassGenerator.java (hunk 1 above)
Comment (marko-bekhta) [reply processor/src/main/java/org/jboss/logging/processor/apt/TranslationClassGenerator.java:153]: Yeah, in the original approach, it was trying to get a resource (and eclipse compiler wants resources to actually exist) while this patch is just creating a new resource (so a potential problem could be that a file with that "random" name actually exists 🙈)

**Comment by jamezp** [comment]
I've gotta be honest here, this seems like an awfully big change for a  bug in a different project. I'm not sure I'd really want to put this fix in here when the fix should really be elsehwere.

**Comment by marko-bekhta** [comment]
Just to be clear 🙈 only the first commit may be of use here (this part https://github.com/jboss-logging/jboss-logging-tools/pull/106/files#diff-aac55a253f897279072fe694e8d4b12f4444838b5d8c121257facd3382320fa1R154-R161). The second one is here temporarily, just in case you want to debug the behaviour of the Eclipse compiler. I agree that the error I've mentioned in the comment above (https://github.com/jboss-logging/jboss-logging-tools/pull/106#issuecomment-2076726534) should be addressed either in `org.jboss.jdeparser` or in the Eclipse compiler itself.

**Comment by jamezp** [comment]
I'm closing this for now. I don't think we want to try to fix a bug in a different project here. If you feel this is unreasonable, please feel free to re-open.

**Comment by marko-bekhta** [comment]
Hey James 😃 yeah, that makes sense, thanks!

For now, Hibernate Search sticks to the 2.14.2 version of `plexus-compiler`, which has no problems. I'll try to see if there are any changes in more recent versions (but that's unrelated to this PR 😄)

**Pull request closed without merging**